doc/source/Tutorials/Scripting/CreatingConfigurationFiles3/index.md

```diff
@@ -0,0 +1,154 @@
+
+We hope that you can use this config as a springboard for adding other custom entries to the node menu. With this solution's simplicity, TDs and artists alike can publish new nodes and add them to the menu.
+
+
+## customNodes.py ##
+
+As with the other startup configs in this tutorial, this one will run in the GUI app. Copy this code to a new `customNodes.py` file in `~/gaffer/startup/gui`:
+
+```eval_rst
+.. code-block:: python
+    :linenos:
+
+    import Gaffer
+    import GafferUI
+    import os
+    
+    def __macbethTexture() :
+    
+    	return Gaffer.Reference( "MacbethTexture" )
+    
+    def __macbethTexturePostCreator( node, menu ) :
+    
+    	node.load(
+    		os.path.expandvars( "$GAFFER_ROOT/resources/references/macbethTexture.grf" )
+    	)
+    
+    nodeMenu = GafferUI.NodeMenu.acquire( application )
+    nodeMenu.append(
+    	path = "/Custom/MacbethTexture",
+    	nodeCreator = __macbethTexture,
+    	postCreator = __macbethTexturePostCreator,
+    	searchText = "MacbethTexture"
+    )
+```
+
+There are three parts to this simple config: a function for adding a node, a function for modifying the node after it's created, and the call that adds these functions to the node menu. Let's take a quick look at each.
+
+
+### The node creation function ###
+
+After `import`ing the necessary modules, the first and simplest step is to declare the menu function. When the user selects the custom entry from the node menu, this is the function that the menu calls. All this function must do is return a node type. It has no relevant arguments for us here.
+
+```eval_rst
+.. code-block:: python
+    :lineno-start: 5
+    
+    def __macbethTexture() :
+    
+    	return Gaffer.Reference( "MacbethTexture" )
+```
+
+There is one small detail we should note. On load, references can modify their containing Reference node's UI metadata, but not the node's name. We therefore have to set the node's name ourselves (line 8, `"MacbethTexture"`). Otherwise, the node would appear in the graph with the generic name _Reference_. The same must be done when renaming any standard node.
+
+
+### The node post-creation function ###
+
+Next, onto the post-creation function, which modifies the recently-created node. It requires two implicit keyword arguments, `node` and `menu`. The first simply refers to the node. The second refers to the node menu, which we will cover in the next section. For this entry, all we need to do is load a reference script into our new Reference node:
+
+```eval_rst
+.. code-block:: python
+    :lineno-start: 9
+    
+    def __macbethTexturePostCreator( node, menu ) :
+    
+    	node.load(
+    		os.path.expandvars( "$GAFFER_ROOT/resources/references/macbethTexture.grf" )
+    	)
+```
+
+```eval_rst
+.. tip::
```

Comment:
@medubelko What do you think to omitting the detail in the tip here? Does it perhaps distract from the focus of this being creating a new node. Would it work if it were simply a reference to the technique/prior tutorial, without the explicit example?